aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -137,6 +140,15 @@ Tensor& _sigmoid_out_cpu(Tensor& result, const Tensor& self) {
   return result;
 }
 
+#ifdef NAMEDTENSOR_ENABLED
+#define PERFORM_NAME_INFERENCE                                  \
+    if (self.is_named()) {                                      \
+      at::internal_set_names_inplace(result, self.names());     \
+    }
+#else
```

Comment:
Would it be better to have a perform_name_inference(result, self) function that checks the namedtensor_enabled guard in the implementation (and is otherwise a no-op)? Seems like then we'd have fewer macros, could be cleaner once we eventually get rid of the namedtensor_enabled guard, we could use in cases where the result/self names differ, and would be clearer what's going on / whats' being passed as arguments in context.

Reply:
Sure, that would look nicer, thanks for the suggestion

Reply:
Changed.